### PR TITLE
Remove non-ascii character from the source code

### PR DESCRIPTION
**Pull request by danielhollas**
Since the line containing the character was commented out, I opted for simply removing it.
For context, see discussion in #237, specifically [this comment](https://github.com/Goddard-Fortran-Ecosystem/pFUnit/issues/237#issuecomment-705888369).

I also took the liberty of fixing a couple of typos in the `README.md` and trying to reword the installation instructions a bit to make it a bit clearer for those (like me) with less CMake experience. Not sure if I succeeded. LMK if you disagree with any of the changes, I'm happy to roll them back.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -181,17 +185,11 @@ the source) to generate the required makefiles.
...
 called `installed` inside the build directory.  This unusual
 default (from a CMake perspective) is due to the fact that
-many/most pFUnit users lack elevated priviges for installing in the
+many/most pFUnit users lack elevated priviliges for installing in the
 usual places.  Most users will want to explicitly override the
 default with CMake's `-DCMAKE_INSTALL_PREFIX=...`
 
-## INSTALLATION
-
-Because many pFUnit users lack permissions to install
```
Comment (danielhollas) [inline README.md:190]: This paragraph was mostly a duplicate of what was already written above.

File: tests/funit-core/Test_AssertString.pf [hunk 2]
```diff
@@ -180,8 +180,6 @@ contains
      call assertEqual( &
           & expected = "i= 1 f= F s=word x=  1.23", &
           & found    = "i= 1  f= F s=word x=  1.23", &
-!          & expected = "i= 1 f= F s=word x=  1.23", &
-!          & found    = “i= 1  f= F s=word x=  1.23", &
```
Comment (danielhollas) [inline tests/funit-core/Test_AssertString.pf:184]: This line contains non-ascii char for double quote.

Comment (tclune) [reply tests/funit-core/Test_AssertString.pf:184]: I should have read more carefully.  I'd not have had you bother if I'd realized that the lines were commented out.  (At least I now know why the compilers were not complaining!)

I'm going to approve, but will probably not bother rolling out a release.

**Review by tclune — APPROVED** [review]
Improvements to the documentation are much appreciated!

File: tests/funit-core/Test_AssertString.pf (hunk 2 above)
Comment (danielhollas) [reply tests/funit-core/Test_AssertString.pf:184]: No problem, happy to help. :-)

**Comment by danielhollas** [comment]
Hmm, not sure what's up with the failing test. :man_shrugging:

**Comment by tclune** [comment]
We have updated the CI on master and are hoping that this PR will now pass the tests.

**Comment by danielhollas** [comment]
Thanks @tclune! :tada:

**Comment by tclune** [comment]
Actually thanks go to @mathomp4  who did all the heavy lifting for the CI.   I just came along after his work and clicked a few approvals.  :-)

**Pull request merged**